sql/core/src/test/scala/org/apache/spark/sql/DataFrameJoinSuite.scala

```diff
@@ -621,4 +621,14 @@ class DataFrameJoinSuite extends QueryTest
     checkAnswer(joined, Row("x", null, null))
     checkAnswer(joined.filter($"new".isNull), Row("x", null, null))
   }
+
+  test("SPARK-47810: replace equivalent expression to <=> in join condition") {
```

Comment:
Let's also add a summary of plan change (before/after) and quantified performance benefit in the PR description

Reply:
Added.